clang/include/clang/Basic/DiagnosticFrontendKinds.td

```diff
@@ -330,6 +330,9 @@ def warn_atomic_op_misaligned : Warning<
   "; the expected alignment (%0 bytes) exceeds the actual alignment (%1 bytes)">,
   InGroup<AtomicAlignment>;
 
+def err_atomic_op_size_zero : Error<
```

Comment:
Rather than introduce a new diagnostic, I think it makes more sense to modify `err_atomic_builtin_must_be_pointer` to say "... must be a pointer %select{to a non-zero-sized object}1 (%0 invalid)".

Reply:
We have around 5–6 diagnostics that all use some variant of ‘address argument to atomic builtin must be X’, which should probably be collapsed into a single diagnostic, but that should probably be a separate pr:
```
def err_atomic_builtin_must_be_pointer : Error<
  "address argument to atomic builtin must be a pointer (%0 invalid)">;
def err_atomic_builtin_must_be_pointer_intptr : Error<
  "address argument to atomic builtin must be a pointer to integer or pointer"
  " (%0 invalid)">;
def err_atomic_builtin_must_be_pointer_intfltptr : Error<
  "address argument to atomic builtin must be a pointer to integer,"
  " floating-point or pointer (%0 invalid)">;
def err_atomic_builtin_pointer_size : Error<
  "address argument to atomic builtin must be a pointer to 1,2,4,8 or 16 byte "
  "type (%0 invalid)">;
```